ad_source/view_sets.py

```diff
@@ -173,3 +181,39 @@ def get(self, request):
             return Response(status=status.HTTP_403_FORBIDDEN)
         logout(request)
         return Response(status=status.HTTP_200_OK)
+
+
+class RewardViewSet(viewsets.ModelViewSet):
+    authentication_classes = [authentication.CsrfExemptSessionAuthentication, BasicAuthentication]
+    queryset = models.Reward.objects.all()
+
+    def create(self, request, *args, **kwargs):
+        task_id = kwargs['task_id']
+        task = get_object_or_404(models.Task, pk=task_id)
+        reward, created = models.Reward.objects.get_or_create(
+            receiver=request.user,
+            task=task,
+            defaults={
+                'sender': task.user,
+                'amount': task.reward_per_click,
+            }
+        )
+        checksummed_sender = Web3.toChecksumAddress(str(reward.sender.username))
+        checksummed_receiver = Web3.toChecksumAddress(str(reward.receiver.username))
+        if created:
+            transaction = contract_instance.functions.forwardRewards(
+                checksummed_receiver,  # To
+                checksummed_sender,  # From
+                task.website_link  # task's website url
+            ).buildTransaction({
+                'chainId': 80001,
```

Comment:
Why `80001`?

Reply:
Matic Mumbai testnet's chainId is 8001. See the documentation below:
https://docs.matic.network/docs/develop/network-details/network/